### DISCUSSION TITLE
Prototype for key override support in custom_config.h

### DISCUSSION THREAD

**Discussion by lilgrassin** [manna-harbour/miryoku#171] [Ideas]
I've completely overhauled my NUM and SYM layers, but still wanted to "shift" between layers using key overrides. It would be nice to be able to use key overrides without modifying manna-harbour_miryoku.c, so I've written up some nice macro spaghetti you might be interested in adding/adapting:

In custom_config.h (U_CUSTOM_SHIFT_OVERRIDE can be exchanged for any of the ko_make fns directly, but I was reducing unnecessary repetition for myself):
```
#define U_CUSTOM_SHIFT_OVERRIDE(_key, _replace, _layer) \
ko_make_with_layers_and_negmods(MOD_MASK_SHIFT, (_key), (_replace), (1 << (_layer)), MOD_MASK_CAG)

#define U_CUSTOM_KEYCODE_OVERRIDE_LIST \
KEYCODE_OVERRIDE_X(comm_base, U_CUSTOM_SHIFT_OVERRIDE, KC_COMM, KC_LPRN, U_BASE) \
KEYCODE_OVERRIDE_X(dot_base, U_CUSTOM_SHIFT_OVERRIDE, KC_DOT, KC_RPRN, U_BASE) \
...
```

In manna-harbour_miryoku.c:
```
#ifdef U_CUSTOM_KEYCODE_OVERRIDE_LIST
#define KEYCODE_OVERRIDE_X(NAME, MAKE_FN, ...) const key_override_t NAME##_key_override = MAKE_FN( __VA_ARGS__ );
U_CUSTOM_KEYCODE_OVERRIDE_LIST
#undef KEYCODE_OVERRIDE_X
#endif

...

const key_override_t **key_overrides = (const key_override_t *[]){
    &capsword_key_override,
#ifdef U_CUSTOM_KEYCODE_OVERRIDE_LIST
#define KEYCODE_OVERRIDE_X(NAME, ...) &NAME##_key_override,
U_CUSTOM_KEYCODE_OVERRIDE_LIST
#undef KEYCODE_OVERRIDE_X
#endif
    NULL
};
```

**Comment by driccio** [comment]
Hi,

Thanks for sharing this. It is exactly what I would ask to @manna-harbour.
I would say it would be nice to do it for all other used features (like tap dance).

Hope that request will be take into account.

Regards
